internal/controller/syncmanager/lifecycle/cluster.go

```diff
@@ -130,7 +133,14 @@ func NewCluster(address string, baseRestConfig *rest.Config) (*Cluster, error) {
 		return newClusterAwareRoundTripper(rt)
 	})
 
+	scheme := runtime.NewScheme()
+
+	if err := kcpdevcorev1alpha1.AddToScheme(scheme); err != nil {
+		return nil, fmt.Errorf("failed to register scheme %s: %w", kcpdevcorev1alpha1.SchemeGroupVersion, err)
+	}
```

Comment:
Remind me, does adding the scheme require it to be available in the endpoint? Will this fail on setups that don't enable this optional flag?

Reply:
To my knowledge, this simply prepares the local client to work with types from the given scheme. This does not change anything about the wire protocol between us and the apiserver. The client will not suddenly do an API discovery and fail if the type doesn't exist.